ambry-cloud/src/main/java/com/github/ambry/cloud/AmbryCloudRequests.java

```diff
@@ -0,0 +1,388 @@
+import com.github.ambry.protocol.AmbryRequests;
+import com.github.ambry.protocol.DeleteRequest;
+import com.github.ambry.protocol.DeleteResponse;
+import com.github.ambry.protocol.GetRequest;
+import com.github.ambry.protocol.GetResponse;
+import com.github.ambry.protocol.PartitionRequestInfo;
+import com.github.ambry.protocol.PartitionResponseInfo;
+import com.github.ambry.protocol.PutRequest;
+import com.github.ambry.protocol.PutResponse;
+import com.github.ambry.protocol.RequestOrResponseType;
+import com.github.ambry.protocol.TtlUpdateRequest;
+import com.github.ambry.protocol.TtlUpdateResponse;
+import com.github.ambry.protocol.UndeleteRequest;
+import com.github.ambry.protocol.UndeleteResponse;
+import com.github.ambry.server.ServerErrorCode;
+import com.github.ambry.server.StoreManager;
+import com.github.ambry.store.IdUndeletedStoreException;
+import com.github.ambry.store.MessageInfo;
+import com.github.ambry.store.Store;
+import com.github.ambry.store.StoreErrorCodes;
+import com.github.ambry.store.StoreException;
+import com.github.ambry.store.StoreGetOptions;
+import com.github.ambry.utils.Utils;
+import io.netty.buffer.ByteBufInputStream;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.EnumSet;
+import java.util.concurrent.atomic.AtomicReference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * The request implementation class for handling requests to Cloud store. All requests from cloud router will be handled
+ * by this class.
+ */
+public class AmbryCloudRequests extends AmbryRequests {
+
+  private static final Logger logger = LoggerFactory.getLogger(AmbryRequests.class);
+
+  /**
+   * @param storeManager {@link StoreManager} object to get stores for replicas
+   * @param requestResponseChannel the {@link RequestResponseChannel} to receive original ambry request and send
+   *                               original ambry response.
+   * @param clusterMap the {@link ClusterMap} of the cluster
+   * @param nodeId the data node Id.
+   * @param registry the {@link MetricRegistry}
+   */
+  public AmbryCloudRequests(StoreManager storeManager, RequestResponseChannel requestResponseChannel,
+      ClusterMap clusterMap, DataNodeId nodeId, MetricRegistry registry) {
+    super(storeManager, requestResponseChannel, clusterMap, nodeId, registry, null, null, null, null, null, null);
+  }
+
+  @Override
+  public void handlePutRequest(NetworkRequest request) throws IOException, InterruptedException {
+    PutRequest receivedRequest;
+
+    // This is a case where handlePutRequest is called when frontends are writing to Azure. In this case, this method
+    // is called by request handler threads running within the frontend router itself. So, the request can be directly
+    // referenced as java objects without any need for deserialization.
+    PutRequest sentRequest =
+        (PutRequest) ((LocalRequestResponseChannel.LocalChannelRequest) request).getRequestInfo().getRequest();
```

Comment:
this is all good, but let's have a check here to make sure the request is actually an instance of LocalChannelRequest, if not, we can return throw an IllegalArgumentException.

Reply:
Done.